File: datasets/pl-keep-code_davinci_001_temp_0.2/HumanEval_98_count_upper.pl
# Given a string s, count the number of uppercase vowels in even indices.
# For example:
# count_upper('aBCdEf') returns 1
# count_upper('abcdefg') returns 0
# count_upper('dBBE') returns 0
sub count_upper {
    my($s) = @_;
    my $count = 0;
    for (my $i = 0; $i < length($s); $i += 2) {
        my $c = substr($s, $i, 1);
        if ($c eq 'A' || $c eq 'E' || $c eq 'I' || $c eq 'O' || $c eq 'U') {
            $count++;
        }
    }
    return $count;
}

use Test::Deep;


sub testhumaneval {
    my $candidate = \&count_upper;
        if(eq_deeply($candidate->("aBCdEf"),1)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("abcdefg"),0)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("dBBE"),0)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("B"),0)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("U"),1)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(""),0)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("EEEE"),2)) {
        print "ok!" }else{
        exit 1;
        }
}

testhumaneval();

File: datasets/r-keep-code_davinci_001_temp_0.2/HumanEval_98_count_upper.r
# Given a string s, count the number of uppercase vowels in even indices.
# For example:
# count_upper('aBCdEf') returns 1
# count_upper('abcdefg') returns 0
# count_upper('dBBE') returns 0
count_upper <- function(s) {
    count <- 0
    for (i in seq(1, length(s), by = 2)) {
        if (s[i] == 'A' | s[i] == 'E' | s[i] == 'I' | s[i] == 'O' | s[i] == 'U') {
            count <- count + 1
        }
    }
    return(count)
}


test_humaneval <- function() {
candidate <- count_upper
    if(!identical(candidate('aBCdEf'), 1)){quit('no', 1)}
    if(!identical(candidate('abcdefg'), 0)){quit('no', 1)}
    if(!identical(candidate('dBBE'), 0)){quit('no', 1)}
    if(!identical(candidate('B'), 0)){quit('no', 1)}
    if(!identical(candidate('U'), 1)){quit('no', 1)}
    if(!identical(candidate(''), 0)){quit('no', 1)}
    if(!identical(candidate('EEEE'), 2)){quit('no', 1)}
}
test_humaneval()